src/plopp/mesh.py

```diff
@@ -76,16 +74,49 @@ def __init__(self,
 
         self._make_mesh(**kwargs)
 
+    def _no_2d_coord(self):
+        return 2 not in (self._data.meta[self._dims['x']].ndim,
+                         self._data.meta[self._dims['y']].ndim)
+
+    def _axis_2d_coord(self):
+        return int(self._data.meta[self._dims['y']].ndim == 2)
+
+    def _maybe_repeat_values(self, array):
+        if self._no_2d_coord():
+            return array.values
+        axis = self._axis_2d_coord()
+        slice_data = (slice(-1), slice(None))[::1 - (2 * axis)]
+        return np.repeat(self._data.data.values, 2, axis=axis)[slice_data]
+
+    def _from_data_array_to_pcolormesh(self):
+        xy = {k: coord_as_bin_edges(self._data, self._dims[k]) for k in 'xy'}
+        z = self._maybe_repeat_values(self._data.data)
+
+        if self._no_2d_coord():
+            return xy['x'].values, xy['y'].values, z
+
+        axis = self._axis_2d_coord()
+        dim_1d, dim_2d = ('y', 'x')[::1 - (2 * axis)]
+        clip = (slice(1, -1), slice(None))[::1 - (2 * axis)]
+        # Broadcast 1d coord to 2d
+        broadcasted_coord = broadcast(xy[dim_1d],
+                                      sizes={
+                                          **xy[dim_2d].sizes,
+                                          **xy[dim_1d].sizes
+                                      }).transpose(self._data.dims).values
+        out = {
+            'xy'[axis - 1]:
+            np.repeat(broadcasted_coord, 2, axis=axis)[clip],
```

Comment:
@SimonHeybrock do you have a good idea on how we could do more of this with scipp instead of numpy? I think the `axis` parameter is a bit bug-prone, would be nice to just use dimension labels...